packages/rx_bloc_cli/mason_templates/bricks/feature_profile/__brick__/lib/feature_profile/blocs/profile_bloc.dart

```diff
@@ -46,37 +45,41 @@ class ProfileBloc extends $ProfileBloc {
   Stream<bool> _mapToIsLoadingState() => loadingState;
 
   @override
-  ConnectableStream<Result<bool>> _mapToSyncNotificationsStatusState() =>
+  ConnectableStream<Result<bool>> _mapToAreNotificationsEnabledState() =>
       Rx.merge([
-        _$setNotificationsEvent.switchMap(
+        _$toggleNotificationsEvent
+            .switchMap((_) =>
+                _notificationService.areNotificationsEnabled().asStream())
+            .switchMap(
+              (areEnabled) => !areEnabled
+                  ? _notificationService
+                      .requestNotificationPermissions()
+                      .asStream()
+                  : Stream.value(areEnabled),
+            )
+            .switchMap((_) =>
+                _notificationService.areNotificationsSubscribed().asStream())
+            .switchMap(
           (subscribePushNotifications) {
-            if (subscribePushNotifications) {
+            if (!subscribePushNotifications) {
```

Comment:
Do you think that subscribing/unsubscribing based on the current status in the notification service it will be a better option ?

Reply:
Yes, done